### PR TITLE
Adds Storybook

### PR DESCRIPTION
**Pull request by chrissrogers**
[See it live](https://recurly.github.io/react-recurly/index.html)

#### Summary
- [Storybook](https://github.com/storybookjs/storybook) as documentation
- Documents all components and hooks presently in `master`
- Adds **Implementation Guide** to the README
- Removes reference sections from README

#### Notes
- Getting started ingests `README.md`
- Some components will populate their prop table from code documentation. This was not possible at this time for all, though, so some rely on manual entry. Manual entry props are defined in `./docs/shared/prop-types.js`
- Similarly, some component descriptions could not readily be parsed by `react-docgen`, which Storybook uses to ingest code commentary. In the meantime, these components are documented with manual entry.
- The IA hierarchy is enforced by the pathname of the story root categories, and then by the `sortOrder` parameter in each stories file
- Visual components use traditional Storybook stories; utility components and hooks use mdx documents.
- The codesandbox demo is included as a documentation page. As yet, no code from the demo is shared with the documentation -- the demo simply loads the codesandbox in an `<iframe>`
- Every `*Element` component uses the `withElements` decorator for stories. I did this to eliminate repetition in the stories, though it may be nice to drive home the necessary `RecurlyProvider` > `Elements` > `*Element` hierarchy.

#### Future desires
- Shared knob values: I'd like to implement a way to persist the `publicKey` so that a reader can input it once and have it readily used across all stories
- All descriptions and prop types ingested from code commentary

### CODE REVIEW

**Comment by drewish** [comment]
That live demo is pretty cool

File: docs/.storybook/preview.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import { version } from '../../package.json';
+
+const theme = create({
+  base: 'light',
+  brandTitle: `react-recurly ${version}`,
+  // brandImage: 'https://recurly.com/img2/brand/wordmark-1@2x.png'
+
+  fontBase: '"Open Sans", Helvetica, sans-serif',
+  appBg: 'rgb(250, 250, 250)',
+  colorSecondary: '#5191b7'
```
Comment (Smolations) [inline docs/.storybook/preview.js]: nit: Mixing and matching color specifiers, eh?  😋

File: package.json [hunk 2]
```diff
@@ -27,8 +27,18 @@
     "@babel/plugin-proposal-class-properties": "^7.8.3",
     "@babel/preset-env": "^7.8.3",
     "@babel/preset-react": "^7.8.3",
+    "@storybook/addon-actions": "^5.3.6",
+    "@storybook/addon-docs": "^5.3.6",
+    "@storybook/addon-knobs": "^5.3.6",
+    "@storybook/addon-links": "^5.3.6",
+    "@storybook/addon-notes": "^5.3.6",
```
Comment (Smolations) [inline package.json]: Didn't see you initialize this addon in the config. Are you still using it or is it a peer dependency for one of the others?

Comment (chrissrogers) [reply package.json]: Yep, good find. I've nixed them

**Pull request merged**